rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -157,17 +159,27 @@ func (s *RollupSyncService) Stop() {
 	}
 }
 
+// ResetStartSyncHeight resets the RollupSyncService to a specific L1 block height
+func (s *RollupSyncService) ResetStartSyncHeight(height uint64) {
+	if s == nil {
+		return
+	}
+
+	s.latestProcessedBlock.Store(height)
+	log.Info("Reset sync service", "height", height)
+}
+
 func (s *RollupSyncService) fetchRollupEvents() {
 	latestConfirmed, err := s.client.getLatestFinalizedBlockNumber()
 	if err != nil {
 		log.Warn("failed to get latest confirmed block number", "err", err)
 		return
 	}
 
-	log.Trace("Sync service fetch rollup events", "latest processed block", s.latestProcessedBlock, "latest confirmed", latestConfirmed)
+	log.Trace("Sync service fetch rollup events", "latest processed block", s.latestProcessedBlock.Load(), "latest confirmed", latestConfirmed)
```

Comment:
We should load `s.latestProcessedBlock` only once at the beginning of the function and store it at the end with CompareAndSwap to avoid synchronization issues with `ResetStartSyncHeight`

If we load multiple times, we might get different values depending on when `ResetStartSyncHeight` updates it.

so it might look something like;


```go
latestProcessedBlock := s.latestProcessedBlock.Load()
updatedLatestProcessedBlock := use(latestProcessedBlock)
s.latestProcessedBlock.CompareAndSwap(latestProcessedBlock, updatedLatestProcessedBlock)
```

if we use `Store` instead of `CompareAndSwap`, we might overwrite the value `ResetStartSyncHeight` might have written.


Same goes for `(s *SyncService) fetchMessages` as well

Reply:
fixed in [014207e](https://github.com/scroll-tech/go-ethereum/pull/1044/commits/014207e66155265e841fc18b59714ff790919dfa).